content/docs/use-cases/experiment-tracking.md

```diff
@@ -0,0 +1,101 @@
+# Machine Learning Experiment Tracking
+
+Making progress on data science projects implies a large number of
+<abbr>experiments</abbr> -- attempts at tuning parameters, improving code,
+trying different data, collecting new metrics, etc. Keeping track of all these
+changes is challenging but essential, as we'll want to review, compare, and
+share them. We'll probably also want to recover the condition that produced
+certain model metrics, and reproduce them if needed.
+
+![](/img/natural-experimentation.png) _Loose experimentation_
```

Comment:
Note ideas here:

```
Algo: XGBoost

Params
trees: 100
learning_rate: 0.3 
max_depth: 6
subsample: 1

Metrics
ROC-AUC: 0.955
Avg. precision: 0.552
Brier score: 0.019

Notes: baseline
```

```
Algo: CatBoost

Params
trees: 100
learning_rate: 0.3 
max_depth: 6
subsample: 1

Metrics
ROC-AUC: 0.957
Avg. precision: 0.572
Brier score: 0.017
```

```
Algo: LightGBM

Params
trees: 100
learning_rate: 0.3 
max_depth: 6
subsample: 1

Metrics
ROC-AUC: 0.947
Avg. precision: 0.553
Brier score: 0.018
```

```
Algo: CatBoost

Params
trees: 100
learning_rate: 0.5
max_depth: 4
subsample: 0.5

Metrics
ROC-AUC: 0.954
Avg. precision: 0.548
Brier score: 0.019

Notes: reduce overfitting
```

```
Algo: CatBoost

Params
trees: 100
learning_rate: 0.3 
max_depth: 6
subsample: 1

Metrics
ROC-AUC: 0.943
Avg. precision: 0.537
Brier score: 0.020

Notes: zero imputation
```

```
Algo: CatBoost

Params
trees: 100
learning_rate: 0.3 
max_depth: 6
subsample: 1

Metrics
ROC-AUC: 0.959
Avg. precision: 0.598
Brier score: 0.015

Notes: mean imputation
```

```
Algo: CatBoost

Params
trees: 100
learning_rate: 0.3 
max_depth: 6
subsample: 1

Metrics
ROC-AUC: 0.958
Avg. precision: 0.605
Brier score: 0.014

Notes: mean imputation & oversampling
```

You could keep combining with similar values (keep in mind that Brier score should go opposite direction of other metrics). If you want more notes, let me know. These are still not that detailed, but feel free to use a subset to keep them shorter.

Reply:
Amazing thanks @dberenbaum ! Still reconsidering that image 😓 but I think this text will be very useful regardless of the final design!

Reply:
p.s. the image incorporates some of this now.